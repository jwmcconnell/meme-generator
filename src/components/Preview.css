.Preview {
  text-align: center;
  img {
    width: 50vw;
    margin: 1rem;
  }
}