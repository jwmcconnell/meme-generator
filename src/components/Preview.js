import React from 'react';
import PropTypes from 'prop-types';
import styles from './Preview.css';

const Preview = ({ upload, url, option, topText, bottomText, fontSize, fontColor }) => {
  const textStyle = {
    fontSize: fontSize + 'px',
    color: fontColor
  };
  
  return (
    <section className={styles.Preview} id="preview">
      <img src={option === 'upload' ? upload : url} />
      <section className={styles.topText} style={textStyle}>{topText}</section>
      <section className={styles.bottomText} style={textStyle}>{bottomText}</section>
    </section>
  );
};

Preview.propTypes = {
  upload: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  option: PropTypes.string.isRequired,
  topText: PropTypes.string.isRequired,
  bottomText: PropTypes.string.isRequired,
  fontSize: PropTypes.number.isRequired,
  fontColor: PropTypes.string.isRequired
};

export default Preview;
