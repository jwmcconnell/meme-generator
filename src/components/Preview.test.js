import React from 'react';
import { shallow } from 'enzyme';
import Preview from './Preview';

describe('Preview component', () => {
  it('renders Preview', () => {
    const state = {
      topText: '',
      bottomText: '',
      fontSize: '26',
      fontColor: '#000000',
      imagePath: '',
      imageUrl: '',
      imageOption: 'upload'
    };
    const wrapper = shallow(
      <Preview 
        upload={state.imagePath} 
        url={state.imageUrl} 
        option={state.imageOption}
        topText={state.topText}
        bottomText={state.bottomText}
        fontSize={state.fontSize}
        fontColor={state.fontColor}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
  
