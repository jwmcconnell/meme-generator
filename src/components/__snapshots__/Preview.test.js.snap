// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview component renders Preview 1`] = `
<section
  className="Preview"
  id="preview"
>
  <img
    src=""
  />
  <section
    className="topText"
    style={
      Object {
        "color": "#000000",
        "fontSize": "16px",
      }
    }
  />
  <section
    className="bottomText"
    style={
      Object {
        "color": "#000000",
        "fontSize": "16px",
      }
    }
  />
</section>
`;
